### PR TITLE
Add Markdown support options

### PR DESCRIPTION
**Pull request by milonoir**
- Add Markdown flag to Message to support disabling Markdown formatting
- Add MarkdownIn field to Attachment to enable Markdown formatting of attachments

Also:
- Fix AuthorLink JSON tag in Attachment

### CODE REVIEW

File: chat/message.go [hunk 1]
```diff
@@ -23,6 +23,9 @@ type Message struct {
 	// Text of the message to send.
 	Text string `json:"text,omitempty"`
 
+	// Markdown enables Markdown support.
+	Markdown bool `json:"mrkdwn,omitempty"`
```
Comment (Tyndie) [inline chat/message.go]: could call just md, like database is db.

Comment (darkliquid) [inline chat/message.go]: I disagree. Attachment has `mrkdwn_in` for its `Markdown` field, so it makes sense for this to remain consistent with that.

File: chat/attachment.go [hunk 2]
```diff
@@ -16,7 +16,7 @@ type Attachment struct {
 	AuthorName string `json:"author_name,omitempty"`
 
 	// AuthorLink is a valid URL that will hyperlink the author_name text mentioned above.
-	AuthorLink string `json:"author_name,omitempty"`
+	AuthorLink string `json:"author_link,omitempty"`
```
Comment (darkliquid) [inline chat/attachment.go]: This is technically a separate fix and should be in a separate commit, as it's not related to the markdown support change.

File: chat/attachment.go [hunk 3]
```diff
@@ -33,6 +33,10 @@ type Attachment struct {
 	// Fields contains optional fields to be displayed in the in a table inside the attachment.
 	Fields []*Field `json:"fields,omitempty"`
 
+	// MarkdownIn enables Markdown support. Valid values are ["pretext", "text", "fields"].
+	// Setting "fields" will enable markup formatting for the value of each field.
+	MarkdownIn []string `json:"mrkdwn_in,omitempty"`
```
Comment (stevenh) [inline chat/attachment.go]: Just spotted this renames the json value lets try to maintain consistency by keeping it the same just converting to camel case as with all the other fields.

File: chat/message.go (hunk 1 above)
Comment (stevenh) [inline chat/message.go]: As above maintain consistency
